desktop/ui/src/main/java/org/datacleaner/windows/ResultWindow.java

```diff
@@ -543,4 +555,17 @@ public void run() {
             }
         });
     }
+
+    public void windowClosed(WindowEvent e) {
+        if (this.getExtendedState() == JFrame.MAXIMIZED_BOTH) {
+            _windowSizePreference.setUserPreferredSize(null, true);
+        } else {
+            _windowSizePreference.setUserPreferredSize(getSize(), false);
+        }
+    }
+
+    @Override
+    protected boolean maximizeWindow() {
```

Comment:
Agreed, but it is coming from the Swing API (overridden method) so I cannot change that.